### ISSUE TITLE
Realistic in silico generation and augmentation of single cell RNA-seq data using Generative Adversarial Neural Networks

### ISSUE THREAD

**Issue by gwaygenomics** [greenelab/deep-review#906]
https://doi.org/10.1101/390153

> A fundamental problem in biomedical research is the low number of observations available, mostly due to a lack of available biosamples, prohibitive costs, or ethical reasons. Augmenting few real observations with generated in silico samples could lead to more robust analysis results and a higher reproducibility rate. Here we propose the use of conditional single cell Generative Adversarial Neural Networks (cscGANs) for the realistic generation of single cell RNA-seq data. cscGANs learn non-linear gene-gene dependencies from complex, multi cell type samples and use this information to generate realistic cells of defined types. Augmenting sparse cell populations with cscGAN generated cells improves downstream analyses such as the detection of marker genes, the robustness and reliability of classifiers, the assessment of novel analysis algorithms, and might reduce the number of animal experiments and costs in consequence. cscGANs outperform existing methods for single cell RNA-seq data generation in quality and hold great promise for the realistic generation and augmentation of other biomedical data types.

**Comment by pierremac** [comment]
Hey Greg (and Greene Lab),

I'm Pierre Machart, one of the authors of the manuscript.
Thanks a lot for your interest in our work! This is a good summary. Just in case in wasn't clear, our (non-conditional) scGAN also uses Batch Normalization (just not the conditional version of it). This being said, it also work very well without it... (It is essential in the conditional version, as it is the sole mechanism to condition the generation of cells.).
The idea of using augmentation to allow for smaller populations, previously unidentified, to be clustered out separately is very interesting (we intend to cover that in future work). However, the way we evaluated the augmentation is fully supervised indeed. We do provide the cluster information a priori (they are just the outcome of the "classic" Louvain clustering in our case). We just show that the ability of a supervised classifier in enhanced by augmenting smaller populations. I hope that clarifies.
There indeed isn't any public code yet but it will definitely happen soon. I will keep you posted about that and our future work (including the exploration of additional applications) if you're interested.

Let me know if you have questions. We are always happy to receive attention and feedback.

**Comment by gwaygenomics** [comment]
Thanks for the response @pierremac - looking forward to the source code and future iterations of the paper! It seems to me that some pretty nifty latent space manipulations can happen fairly quickly with a well trained model

**Comment by pierremac** [comment]
One of the many things we intend to look into, indeed!
We also just came across this very recent manuscript: https://www.biorxiv.org/content/early/2018/07/30/262501
Their aim is not the realistic simulation of cells but it is closely related to our work and you will maybe appreciate that they put a bit more emphasis on latent space manipulations!

**Comment by gwaygenomics** [comment]
Fantastic, thanks for pointing to. Want to summarize in a new issue? Contributions are welcome :)

**Comment by pierremac** [comment]
The source code is finally available.
Please find it here : https://github.com/imsb-uke/scGAN
We'll be very happy to get your feedback!
